plugins/topology/src/components/Topology/TopologyComponent.tsx

```diff
@@ -4,18 +4,26 @@ import '@patternfly/react-core/dist/styles/base.css';
 import { useTheme } from '@material-ui/core/styles';
 import '@patternfly/patternfly/patternfly-theme-dark.css';
 import { TopologyWorkloadView } from './TopologyWorkloadView';
-import { K8sResourcesContext } from '../../hooks/K8sResourcesContext';
-import { useAllWatchResources } from '../../hooks/useAllWatchResources';
+import {
+  K8sResourcesClusterContext,
+  K8sResourcesClustersContext,
+} from '../../hooks/K8sResourcesContext';
 
 import './TopologyComponent.css';
+import { useEntity } from '@backstage/plugin-catalog-react';
+import { useKubernetesObjects } from '@backstage/plugin-kubernetes';
+import { useK8sResourcesClusters } from '../../hooks/useK8sResourcesClusters';
 
 const THEME_DARK = 'dark';
 const THEME_DARK_CLASS = 'pf-theme-dark';
 
 export const TopologyComponent = () => {
+  const { entity } = useEntity();
+  const k8sObjectsResponse = useKubernetesObjects(entity);
```

Comment:
will be nice to move this to a custom hook which we would have for context, WDYT?

Reply:
So you are saying have a hook which then should be used here or in the hooks I want the `k8sObjectsResponse` response in? If it the later then the api call will be made more than once.

Reply:
i was thinking of just one context provider which can return what all need and consumer can pick the data as needed